sw/device/lib/dif/dif_padctrl.c

```diff
@@ -0,0 +1,180 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "sw/device/lib/dif/dif_padctrl.h"
+
+#include "padctrl_regs.h"
+#include "sw/device/lib/base/mmio.h"
+
+#define PADCTRL_REGISTER_WIDTH_IN_BYTES (PADCTRL_PARAM_REG_WIDTH / 8)
+
+// This is the value written to a whole attribute register, it is not a per-pad
+// value so desired default attribute setting must be replicated per attribute
+// field in the attribute register.
+#define PADCTRL_ATTR_DEFAULT 0
+#define PADCTRL_ATTR_FIELD_MASK PADCTRL_MIO_PADS0_ATTR0_MASK
+
+/**
+ * Checks if an attribute field contains conflicting attributes
+ */
+static bool dif_padctrl_conflicting_attributes(uint32_t attrs) {
```

Comment:
Nit: you can drop the `dif_padctrl` in `static` functions, since those are not visible outside this header.

Reply:
done